File: backend/src/main/java/com/sms/service/StudentService.java
package com.sms.service;

import com.sms.entity.Student;
import com.sms.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class StudentService {
    
    @Autowired
    private StudentRepository studentRepository;
    
    // Get all students
    public List<Student> getAllStudents() {
        return studentRepository.findAll();
    }
    
    // Get student by ID
    public Optional<Student> getStudentById(Long id) {
        return studentRepository.findById(id);
    }
    
    // Create new student
    public Student createStudent(Student student) {
        if (studentRepository.existsByEmail(student.getEmail())) {
            throw new RuntimeException("Email already exists");
        }
        return studentRepository.save(student);
    }
    
    // Update student
    public Student updateStudent(Long id, Student studentDetails) {
        Student student = studentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Student not found"));
        
        student.setName(studentDetails.getName());
        student.setEmail(studentDetails.getEmail());
        student.setPhoneNumber(studentDetails.getPhoneNumber());
        student.setCourse(studentDetails.getCourse());
        student.setEnrollmentNumber(studentDetails.getEnrollmentNumber());
        student.setGpa(studentDetails.getGpa());
        student.setAddress(studentDetails.getAddress());
        
        return studentRepository.save(student);
    }
    
    // Delete student
    public void deleteStudent(Long id) {
        Student student = studentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Student not found"));
        studentRepository.delete(student);
    }
    
    // Search students by name
    public List<Student> searchByName(String name) {
        return studentRepository.findByNameContainingIgnoreCase(name);
    }
    
    // Filter by course
    public List<Student> filterByCourse(String course) {
        return studentRepository.findByCourseContainingIgnoreCase(course);
    }
}

File: frontend/src/services/StudentService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/students';

const StudentService = {
  getAllStudents: () => {
    return axios.get(API_URL);
  },

  getStudentById: (id) => {
    return axios.get(`${API_URL}/${id}`);
  },

  createStudent: (student) => {
    return axios.post(API_URL, student);
  },

  updateStudent: (id, student) => {
    return axios.put(`${API_URL}/${id}`, student);
  },

  deleteStudent: (id) => {
    return axios.delete(`${API_URL}/${id}`);
  },

  searchByName: (name) => {
    return axios.get(`${API_URL}/search/${name}`);
  },

  filterByCourse: (course) => {
    return axios.get(`${API_URL}/filter/course/${course}`);
  }
};

export default StudentService;
